nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenHTTP.java

```diff
@@ -373,6 +386,14 @@ public void createHttpServer(final ProcessContext context) throws Exception {
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) {
+        try {
+            if (!initialized.get()) {
```

Comment:
If the processor is scheduled with multiple threads, we could have multiple threads entering `createHttpServerFromService` simultaneously. Each can then create a new server, resulting in multiple Jetty Servers being created and started but only one ever getting stored as `this.server` so we'd be leaking the server resource. It's important that the method itself be marked as `synchronized` or that a `synchronized` block is used here to guard against this.